internal/predicates/osbuildconfig.go

```diff
@@ -29,5 +29,9 @@ func (OSBuildConfigChangedPredicate) Update(e event.UpdateEvent) bool {
 	if newConfig.Status.LastTemplateResourceVersion != nil && newConfig.Status.CurrentTemplateResourceVersion != nil {
 		templateChanged = *newConfig.Status.LastTemplateResourceVersion != *newConfig.Status.CurrentTemplateResourceVersion
 	}
+
+	templateTriggerEnabled := newConfig.Spec.Triggers.TemplateConfigChange == nil || *newConfig.Spec.Triggers.TemplateConfigChange
```

Comment:
If we're checking if template change should trigger a new build, we should also check if a new config should trigger a new build (for the generationChanged flag)

Reply:
:+1:

Reply:
Done